pkg/clusterresource/builder.go

```diff
@@ -133,6 +146,32 @@ func (o *Builder) Validate() error {
 		return fmt.Errorf("must set serving cert key to use with serving cert")
 	}
 
+	if len(o.HTTPProxy) > 0  || len(o.HTTPSProxy) > 0  {
+		if o.HTTPProxy == "" && o.HTTPSProxy == "" {
+			return fmt.Errorf("must set HTTPProxy or HTTPSProxy")
```

Comment:
If above assumptions is correct, this should probably reflect that, something like "cannot set both HTTPProxy and HTTPSProxy".